qubesbuilder/component.py

```diff
@@ -29,9 +29,27 @@
 import yaml
 from packaging.version import Version, InvalidVersion
 
-from qubesbuilder.common import sanitize_line
+from qubesbuilder.common import sanitize_line, STAGES
 from qubesbuilder.exc import ComponentError
 
+FORBIDDEN_PATTERNS = [".."]
+for s in STAGES:
+    FORBIDDEN_PATTERNS += [f".{s}.yml", f".{s}.yaml"]
```

Comment:
This is probably necessary too, but I think it should check for filename conflicts with anything that is in `copy_out`. This includes names of subpackages and any other file list that is determined dynamically (extend `is_filename_valid()`? and verify if all the places that dynamically build a filename use it). In fact `source_{rpm,deb}` may require that all subpackages have proper extension (`.deb`, `.rpm` etc).